consensus/common/src/main/java/org/hyperledger/besu/consensus/common/CombinedProtocolScheduleFactory.java

```diff
@@ -52,7 +52,9 @@ public BftProtocolSchedule create(
           Optional.ofNullable(forkSpecs.higher(spec)).map(ForkSpec::getBlock);
       protocolSchedule.getScheduledProtocolSpecs().stream()
           .filter(protocolSpecMatchesConsensusBlockRange(spec.getBlock(), endBlock))
-          .forEach(s -> combinedProtocolSchedule.putBlockNumberMilestone(s.milestone(), s.spec()));
+          .forEach(
+              s ->
+                  combinedProtocolSchedule.putBlockNumberMilestone(s.fork().milestone(), s.spec()));
```

Comment:
The HardFork layer of indirection:
`s.fork().milestone()` instead of `s.milestone()`

See https://github.com/hyperledger/besu/pull/5536#discussion_r1251804232